spring-integration-file/src/main/java/org/springframework/integration/file/config/FileReadingMessageSourceFactoryBean.java

```diff
@@ -128,20 +132,19 @@ protected FileReadingMessageSource createInstance() {
 	}
 
 	private void initSource() {
-		boolean comparatorSet = this.comparator != null;
-		boolean queueSizeSet = this.queueSize != null;
-		if (comparatorSet) {
-			if (queueSizeSet) {
+		if (this.comparator != null) {
+			if (this.queueSize != null) {
 				logger.warn("'comparator' and 'queueSize' are mutually exclusive. Ignoring 'queueSize'");
 			}
 			this.source = new FileReadingMessageSource(this.comparator);
 		}
-		else if (queueSizeSet) {
+		else if (this.queueSize != null) {
 			this.source = new FileReadingMessageSource(this.queueSize);
 		}
 		else {
 			this.source = new FileReadingMessageSource();
 		}
+		Assert.notNull(this.directory, "The 'directory' must be provided.");
```

Comment:
This can be removed because the setDirectory already checks for null.

Reply:
Right. And what if `setDirectory()` was not called?